lib/em-http/http_connection.rb

```diff
@@ -198,7 +198,7 @@ def post_init
       end
 
       @p.on_message_complete = proc do
-        if !client.continue?
+        if client && !client.continue?
```

Comment:
🗒️ This code was introduced in 3a9b96c2f31f29c112a11b934fee7d40cafd4dcc but the corresponding spec never worked. It failed with this error:

```
NoMethodError:
  undefined method `continue?' for nil:NilClass
```

This change fixes the nil error so that the spec passes as intended.